internal/pkg/views/notifications.go

```diff
@@ -0,0 +1,175 @@
+package views
+
+import (
+	"context"
+	"fmt"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/style"
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jroimartin/gocui"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/eventing"
+)
+
+func init() {
+	pendingDeletes = []pendingDelete{}
+}
+
+type pendingDelete struct {
+	display string
+	url     string
+}
+
+var pendingDeletes []pendingDelete
+var deleteMutex sync.Mutex // ensure delete occurs only once
+var gui *gocui.Gui
+
+// AddPendingDelete queues deletes for
+// delete once confirmed
+func AddPendingDelete(display, url string) {
+	if url == "" {
+		eventing.SendStatusEvent(eventing.StatusEvent{
+			Failure: true,
+			Message: "Item `" + display + "` doesn't support delete",
+			Timeout: time.Second * 5,
+		})
+		return
+	}
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		for _, i := range pendingDeletes {
+			if i.url == url {
+				eventing.SendStatusEvent(eventing.StatusEvent{
+					Failure: true,
+					Message: "Item already `" + display + "` in pending delete list",
+					Timeout: time.Second * 5,
+				})
+				return nil
+			}
+		}
+
+		pendingDeletes = append(pendingDeletes, pendingDelete{
+			url:     url,
+			display: display,
+		})
+
+		return nil
+	})
+}
+
+// ConfirmDelete delete all queued/pending deletes
+func ConfirmDelete() {
+	gui.Update(func(g *gocui.Gui) error {
+		deleteMutex.Lock()
+		defer deleteMutex.Unlock()
+
+		pending := pendingDeletes
+		event, _ := eventing.SendStatusEvent(eventing.StatusEvent{
+			InProgress: true,
+			Message:    "Starting to delete items",
+			Timeout:    time.Second * 15,
+		})
+
+		for _, i := range pending {
+			_, err := armclient.DoRequest(context.Background(), "DELETE", i.url)
+			if err != nil {
+				event.Failure = true
+				event.Message = "Failed to delete `" + i.display + "` with error:" + err.Error()
+				event.Update()
+
+				// In the event that a delete fails in the
+				// batch of pending deletes lets give up on the rest
+				// as something might have gone wrong and best
```

Comment:
Understandable. I'm in 2 minds about this. It could be dangerous to change this behaviour later if users get used to it and as it stands it implies the delete order is the same as the pending queue order which may not always be the case? i.e. if you move to batch or map. Likely not an issue as it'd probably be rare for a user to pick up on the subtleties but just thought I'd call it out.

Reply:
Yeah good point. Having looked more at the Batch API it can't handle more than 20 requests at a time so feels like there would always be some form of paging going on. 

My gut at the moment is to give up on the Batch API and leave this behavior as is - in my personal use case I'm usually only deleting 3/4 things at a time but I may not be normal.